app/fields/date_field.rb

```diff
@@ -35,4 +35,14 @@ def self.from_persisted_value(date)
     end
   end
 
+  def set_error_message_lookup_proc!(error_proc)
+    @error_message_lookup_proc = error_proc
+  end
+
+  private
+
+  def error_message_lookup_proc
+    @error_message_lookup_proc || ->(a){ a }
```

Comment:
It should only get called once, when trying to submit the form, but it is a valid optimisation nonetheless.